benchmarks/bench_plot_nmf.py

```diff
@@ -1,168 +1,420 @@
 """
 Benchmarks of Non-Negative Matrix Factorization
 """
+# Author : Tom Dupre la Tour <tom.dupre-la-tour@m4x.org>
+# License: BSD 3 clause
 
 from __future__ import print_function
-
-from collections import defaultdict
-import gc
 from time import time
-
-import six
+import sys
+import warnings
+import numbers
 
 import numpy as np
-from scipy.linalg import norm
+import matplotlib.pyplot as plt
+import pandas
+
+from sklearn.utils.testing import ignore_warnings
+from sklearn.feature_extraction.text import TfidfVectorizer
+from sklearn.decomposition.nmf import NMF
+from sklearn.decomposition.nmf import _initialize_nmf
+from sklearn.decomposition.nmf import _beta_divergence
+from sklearn.decomposition.nmf import INTEGER_TYPES, _check_init
+from sklearn.externals.joblib import Memory
+from sklearn.exceptions import ConvergenceWarning
+from sklearn.utils.extmath import fast_dot, safe_sparse_dot, squared_norm
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted, check_non_negative
+
+
+mem = Memory(cachedir='.', verbose=0)
 
-from sklearn.decomposition.nmf import NMF, _initialize_nmf
-from sklearn.datasets.samples_generator import make_low_rank_matrix
-from sklearn.externals.six.moves import xrange
 
+###################
+# Start of _PGNMF #
+###################
```

Comment:
If you keep it maybe you can add a comment to say where comes from that codes et when it was removed of sklearn ?